src/Features/Core/Portable/Intents/IIntentProvider.cs

```diff
@@ -0,0 +1,15 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Features.Intents
+{
+    internal interface IIntentProvider
+    {
+        Task<Solution?> ComputeActionForIntentAsync(Document documentBeforeIntent, TextSpan selection, Document currentDocument, string? serializedIntentData, CancellationToken cancellationToken);
```

Comment:
'priorDocument' 'currentDocument'

Reply:
`ComputeActionForIntentAsync` is odd, since we're not getting an Action back.  How about just ComputeIntent?